src/modules/feature-modules/innovator/guards/manage-innovation.guard.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { InnovationsService } from '@modules/shared/services/innovations.service';
+import { AuthenticationStore } from '@modules/stores';
+import { catchError, map, Observable, of } from 'rxjs';
+
+@Injectable()
+export class ManageInnovationGuard implements CanActivate {
+
+  constructor(
+    private router: Router,
+    private authenticationStore: AuthenticationStore,
+    private innovationsService: InnovationsService
+  ) { }
+
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot,
+    ): Observable<boolean> {
+
+    return this.innovationsService.getInnovationInfo(route.params.innovationId).pipe(
```

Comment:
It is necessary because if the user navigates to the manage-innovation through the browser, the Resolver that populates the innovation in the context is only called after passing through the manage-innovation route Guard